spec-tests/src/test/java/org/candlepin/spec/imports/ImportErrorSpecTest.java

```diff
@@ -0,0 +1,205 @@
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+
+package org.candlepin.spec.imports;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatNoException;
+import static org.candlepin.spec.bootstrap.assertions.StatusCodeAssertions.assertBadRequest;
+import static org.candlepin.spec.bootstrap.assertions.StatusCodeAssertions.assertConflict;
+
+import org.candlepin.dto.api.client.v1.ImportRecordDTO;
+import org.candlepin.dto.api.client.v1.OwnerDTO;
+import org.candlepin.dto.api.client.v1.PoolDTO;
+import org.candlepin.dto.api.client.v1.UpstreamConsumerDTO;
+import org.candlepin.dto.api.client.v1.UserDTO;
+import org.candlepin.spec.bootstrap.assertions.OnlyInStandalone;
+import org.candlepin.spec.bootstrap.client.ApiClient;
+import org.candlepin.spec.bootstrap.client.ApiClients;
+import org.candlepin.spec.bootstrap.client.SpecTest;
+import org.candlepin.spec.bootstrap.data.builder.Consumers;
+import org.candlepin.spec.bootstrap.data.builder.Export;
+import org.candlepin.spec.bootstrap.data.builder.ExportGenerator;
+import org.candlepin.spec.bootstrap.data.builder.Owners;
+import org.candlepin.spec.bootstrap.data.util.Importer;
+import org.candlepin.spec.bootstrap.data.util.UserUtil;
+
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.parallel.Execution;
+import org.junit.jupiter.api.parallel.ExecutionMode;
+
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * Spec test suite focused on verification of various errors occurring dung import.
+ *
+ * Requires per_class lifecycle to enable async tests to override importer.
+ */
+@SpecTest
+@OnlyInStandalone
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class ImportErrorSpecTest {
+
+    private ApiClient admin;
+    private OwnerDTO owner;
+    private ApiClient userClient;
+    /**
+     * This is a separate import that is older than the other export. Not to be
+     * confused with older version of the same export.
+     */
+    private Export otherExportOld;
+    private Export exportOld;
+    private Export export;
+    private Importer importer;
+
+    @BeforeAll
+    public void beforeAll() {
+        admin = ApiClients.admin();
+        owner = admin.owners().createOwner(Owners.random());
+        UserDTO user = UserUtil.createUser(admin, owner);
+        userClient = ApiClients.trustedUser(user.getUsername());
```

Comment:
Can we replace all `ApiClients.trustedUser(username)` with `ApiClients.basic(username, password)` through out this PR (and for all other spec tests from now on)? The ApiClients.trustedUser and ApiClients.trustedConsumer should never be used in spec tests (aside from the few tests that explicitly test those authentication methods, e.g. TrustedConsumerAuthSpecTest & TrustedUserAuthSpecTest).

Reply:
Done.